gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -275,6 +279,7 @@ static void addJobStatusToStateStore(org.apache.gobblin.configuration.State jobS
       modifyStateIfRetryRequired(jobStatus);
       stateStore.put(storeName, tableName, jobStatus);
       if (isNewStateTransitionToFinal(jobStatus, states)) {
+        dagActionStore.addDagAction(flowGroup, flowName, flowExecutionId, jobName, DagActionStore.DagActionType.REEVALUATE);
         eventProducer.emitObservabilityEvent(jobStatus);
```

Comment:
the fact that you must add mocking for the `DagActionStore` just to keep existing tests working that call this `static` method, suggests the method is doing too much.  instead, how about returning `Optional<JobStatus>` that is present only in cases where `isNewStateTransitionToFinal`?

that way `dagActionStore` and `eventProducer` are only called directly within `processMessage`